source/util/parse_number.cpp

```diff
@@ -0,0 +1,174 @@
+// Copyright (c) 2016 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "util/parse_number.h"
+
+#include <functional>
+#include <iomanip>
+#include <sstream>
+#include <string>
+#include <tuple>
+
+#include "util/hex_float.h"
+
+namespace spvutils {
+std::tuple<EncodeNumberStatus, std::string> ParseAndEncodeIntegerNumber(
```

Comment:
It is wasteful to construct a std::string on every use of this function.  Under normal operation the error will not be generated.   Although the code as proposed is the idiomatic way to do this in Go, I'd prefer to pass a pointer-to-string as the last argument, and it is filled in only on error.  Accept nullptr as meaning "I don't care about the error details, just throw them away".